baseapp/baseapp.go

```diff
@@ -586,6 +593,18 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (gInfo sdk.
 		msCache.Write()
```

Comment:
since we know there is no error we can drastically cleanup the code and move the events appending here:

```go
if len(events) > 0:
    result.Events = append(events.ToABCIEvents(), result.Events...)
```